python/bigipconfigdriver.py

```diff
@@ -507,32 +739,50 @@ def _create_client_ssl_profile(bigip, profile):
                                   defaultsFrom=None)
     except Exception as err:
         log.error("Error creating SSL profile: %s" % err.message)
+        incomplete = 1
 
+    return incomplete
+
+
+def _delete_client_ssl_profiles(mgmt, partition, config):
+    incomplete = 0
+
+    try:
+        profiles = mgmt.tm.ltm.profile.client_ssls.get_collection(
+            requests_params={'params': '$filter=partition+eq+%s'
+                             % partition})
+    except Exception as err:
+        log.error("Error reading profiles from BIG-IP: %s" % err.message)
+        incomplete += 1
+        return incomplete
 
-def _delete_client_ssl_profiles(bigip, config):
     if 'customProfiles' not in config:
-        # delete any profiles in managed partitions
-        for partition in bigip._partitions:
-            for prof in bigip.ltm.profile.client_ssls.get_collection(
-                requests_params={'params': '$filter=partition+eq+%s'
-                                 % partition}):
+        # delete any profiles in managed partition
+        for prof in profiles:
+            try:
                 prof.delete()
+            except Exception as err:
+                log.error("Error deleting SSL profile: %s" % err.message)
+                incomplete += 1
     else:
         # delete profiles no longer in our config
-        for partition in bigip._partitions:
-            for prof in bigip.ltm.profile.client_ssls.get_collection(
-                requests_params={'params': '$filter=partition+eq+%s'
-                                 % partition}):
-                if not any(d['name'] == prof.name and
-                           d['partition'] == partition
-                           for d in config['customProfiles']):
+        for prof in profiles:
+            if not any(d['name'] == prof.name and
+                       d['partition'] == partition
+                       for d in config['customProfiles']):
+                try:
                     prof.delete()
+                except Exception as err:
+                    log.error("Error deleting SSL profile: %s" % err.message)
+                    incomplete += 1
+
+    return incomplete
 
 
 class ConfigHandler():
```

Comment:
Cool, I think the has-a approach worked out well (at least it's more clear to me)